tracer/transport_test.go

```diff
@@ -59,3 +58,57 @@ func TestTracesAgentIntegration(t *testing.T) {
 		assert.Equal(200, response.StatusCode)
 	}
 }
+
+func TestAPIDowngrade(t *testing.T) {
+	assert := assert.New(t)
+	transport := newHTTPTransport("http://localhost:7777/v0.0/traces")
+
+	// if we get a 404 we should downgrade the API
+	traces := getTestTrace(2, 2)
+	response, err := transport.Send(traces)
```

Comment:
Where's the server (or mock) listening to the traces?

When I run `go test`, I get

```
--- FAIL: TestAPIDowngrade (0.00s)
        Error Trace:    transport_test.go:69
	Error:		Expected nil, but got: &url.Error{Op:"Post", URL:"http://localhost:7777/v0.0/traces", Err:(*net.OpError)(0xc4201212c0)}
		
        Error Trace:    transport_test.go:71
	Error:		Not equal: 200 (expected)
			        != 0 (actual)
```

Similar problems with TestEncoderDowngrade and TestTracesAgentIntegration. From CI logs, I am not sure that these tests are run in the first place...